cheroot/test/test_core.py

```diff
@@ -333,6 +333,7 @@ def test_malformed_request_line(
     c.close()
 
 
+@pytest.mark.xfail(reason="h11 normalizes the method and ignores non-capitalized methods")
```

Comment:
Contribution to H11 needed?

Reply:
This appears to be by design? @njsmith

Reply:
h11 seems to be happy to work with lowercase methods here, so I think the diagnosis is wrong?

```
In [2]: import h11                                                                        

In [3]: c = h11.Connection(our_role=h11.CLIENT)                                           

In [5]: h11.Request(method="get", headers=[("host", "example.com")], target="/")          
Out[5]: Request(method=b'get', target=b'/', headers=[(b'host', b'example.com')], http_version=b'1.1')

In [6]: c.send(_5)                                                                        
Out[6]: b'get / HTTP/1.1\r\nhost: example.com\r\n\r\n'

In [7]: s = h11.Connection(our_role=h11.SERVER)                                           

In [8]: s.receive_data(_6)                                                                

In [9]: s.next_event()                                                                    
Out[9]: Request(method=b'get', target=b'/', headers=[(b'host', b'example.com')], http_version=b'1.1')
```